MediaAdminBundle/Transformer/FolderTreeTransformer.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace OpenOrchestra\MediaAdminBundle\Transformer;
+
+use OpenOrchestra\BaseApi\Facade\FacadeInterface;
+use OpenOrchestra\BaseApi\Transformer\AbstractTransformer;
+use OpenOrchestra\Media\Model\FolderInterface;
+use OpenOrchestra\MediaAdminBundle\Facade\FolderTreeFacade;
+
+/**
+ * Class FolderTreeTransformer
+ */
+class FolderTreeTransformer extends AbstractTransformer
+{
+    /**
+     * @param $folderCollection
+     *
+     * @return FolderTreeFacade
+     */
+    public function transform($folderCollection)
+    {
+        if (empty($folderCollection)) {
+            return array();
+        }
+
+        $facade = $this->newFacade();
+
+        if ($folderCollection instanceof FolderInterface) {
```

Comment:
Cause there are many root folders and not only one. So we can't start creating the tree from a FolderInterface who don't exist but with a collection of the root FolderInterface. 
The first node of the created tree facade will be folder=null and children=collection of the root folders